File: MoneyFlowUI/src/app/shared/dtos/DTO_CardsPage.ts
import { DTO_Card } from './DTO_Card';

export class DTO_CardsPage {
    cards: DTO_Card[] = [];
    totalCardsCount = 0;
    activeCardsCount = 0;
    creditCardsCount = 0;
    debitCardsCount = 0;
    latestCardName = 'Sem dados';
    latestCardCreatedAt: string | null = null;
}

File: MoneyFlowAPI/Application/DTOs/DTO_CardsPage.cs
namespace MoneyFlowAPI.Application.DTOs
{
    public class DTO_CardsPage
    {
        public List<DTO_Card> Cards { get; set; } = new();

        public int TotalCardsCount { get; set; }

        public int ActiveCardsCount { get; set; }

        public int CreditCardsCount { get; set; }

        public int DebitCardsCount { get; set; }

        public string LatestCardName { get; set; } = "Sem dados";

        public DateTime? LatestCardCreatedAt { get; set; }
    }
}
